suzieq/poller/worker/nodes/node.py

```diff
@@ -494,7 +499,8 @@ async def _get_device_type_hostname(self):
         # setup time. show version works on most networking boxes and
         # hostnamectl on Linux systems. That's all we support today.
         await self._exec_cmd(self._parse_device_type_hostname,
-                             ["show version",
+                             ["show system",
+                              "show version",
```

Comment:
This really should be fixed with a durable solution.

There is a problem with the ´only_one´ parameters on the cmd execution in function ´_get_device_type_hostname´ for the device discovery. If an unknown command is sent to the device, status 0 and an error message in standard output get returned. So when the correct command for that type of device is not used first, the device gets discovered as unknown.

The program can also never enter the elif function because only one command from the list is sent if the return status is 0 (even with an error return):

```
        elif (len(output) > 1) and (output[1]["status"] == 0):
            devtype = 'unsupported'
            data = output[1]["data"]
            if data:
                if "Cumulus Linux" in data:
                    devtype = "cumulus"
                else:
                    devtype = "linux"

                version_str = data
                # Hostname is the last line of the output
                if len(data.strip()) > 0:
                    hostname = data.splitlines()[-1].strip()
```

Sample terminal output on unknown cli command (with return status code 0):

```
net-swi-001> show version
                     ^
ERROR: Invalid entry: "version"
```

Reply:
> This really should be fixed with a durable solution.

I agree. What you propose, checking for a command failure using both the return code AND a regex for the output, seems the best way to go.

For this first phase I'd go for supporting Alcatel-Lucent only supplying `devtype` as `aos` in the inventory, we can implement the new way of determining a failed command separately and then come back to having the autodiscovery.

> Sample terminal output on unknown cli command (with return status code 0):
> 
> net-swi-001> show version
>                      ^
> ERROR: Invalid entry: "version"

Does this appear in the stdout or stderr?

Reply:
Terminal output from cli command, so stdout.